src/CattyTests/Formula/FormulaElementTest.swift

```diff
@@ -53,4 +53,36 @@ final class FormulaElementTest: XCTestCase {
         }
         internTokenList.removeAllObjects()
     }
+
+    func testIsSingleNumberFormulaMethod() {
+        var isSingleNumberFormula: Bool
+        let element = FormulaElement(elementType: ElementType.NUMBER,
+                                     value: nil,
+                                     leftChild: FormulaElement(double: 1),
+                                     rightChild: nil,
+                                     parent: nil)
+
+        if let formulaElement = element {
+            isSingleNumberFormula = formulaElement.isSingleNumberFormula()
+            XCTAssertTrue(isSingleNumberFormula)
```

Comment:
Please do not unwrap optionals in test cases like this. This code path is not executed once the optional is nil. Hence, the assert statement will not be executed and the test will pass even though it should not (i.e. when `element` is nil). Your approach would work fine if you add a fail statement (i.e. `XCTFail`) in a separate else-branch to check the extra state of optionals. Nonetheless, this will increase complexity and make the code less readable. The best solution I can think of to compare optional bools in tests is:

`XCTAssertEqual(element?.isSingleNumberFormula(), true)`

This single line statement will cover all possible states of optionals and fail if (a) `isSingleNumberFormula` returns a wrong bool; or, (b) `element` is nil.

At least as important as this: please add a negative test scenario as well (asserting that e.g. 2 is not a "single number formula").